coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+
+                :param output:
+                    The output of the program. This can be either a single
+                    string or a sequence of strings.
+                :param filename:
+                    The filename of the file currently being corrected.
+                :param file:
+                    The contents of the file currently being corrected.
+                :param output_regex:
+                    The regex to parse the output with. It should use as many
+                    of the following named groups (via ``(?P<name>...)``) to
+                    provide a good result:
+
+                    - line - The line where the issue starts.
+                    - column - The column where the issue starts.
+                    - end_line - The line where the issue ends.
+                    - end_column - The column where the issue ends.
+                    - severity - The severity of the issue.
+                    - message - The message of the result.
+                    - origin - The origin of the issue.
+
+                    The groups ``line``, ``column``, ``end_line`` and
+                    ``end_column`` don't have to match numbers only, they can
+                    also match nothing, the generated ``Result`` is filled
+                    automatically with ``None`` then for the appropriate
+                    properties.
+                :param severity_map:
+                    A dict used to map a severity string (captured from the
+                    ``output_regex`` with the named group ``severity``) to an
+                    actual ``coalib.results.RESULT_SEVERITY`` for a result.
+                :return:
+                    An iterator returning results.
+                """
+                if isinstance(output, str):
+                    output = (output,)
+
+                for string in output:
+                    for match in re.finditer(output_regex, string):
+                        yield self._convert_output_regex_match_to_result(
+                            match, filename, severity_map=severity_map)
+
+            if options["output_format"] is None:
+                # Check if user supplied a `process_output` override.
+                if not (hasattr(klass, "process_output") and
+                        callable(klass.process_output)):
+                    raise ValueError("`process_output` not provided by given "
+                                     "class.")
+                # No need to assign to `process_output` here, the class mixing
+                # below automatically does that.
+            else:
+                # Prevent people from accidentally defining `process_output`
+                # manually, as this would implicitly override the internally
+                # set-up `process_output`.
+                if hasattr(klass, "process_output"):
+                    raise ValueError("`process_output` is used by given class,"
```

Comment:
`Found `process_output()` defined and `output_format={}` in class {}..format(<output_format>, <class_name>)`

Seems more concise and easier to read